### PR TITLE
Circle CI build job for ARM64

### LINKED ISSUE #1891: Circle CI: Enable testing for ARM64 platform
Hi Team,
I have been working on running and testing Falco on arm64 platform. Circle CI allows to run jobs on arm64 architecture as can be seen [here](https://circleci.com/execution-environments/arm/). 

Would you be interested to enable testing for arm64 platform on Circle CI? 
Would be happy to help with it.

### PR DESCRIPTION
**Pull request by odidev**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area build

**What this PR does / why we need it**:
Added Circle CI job to build Falco for ARM64 platform. 

**Which issue(s) this PR fixes**:

Fixes #1891 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,5 +1,29 @@
-version: 2
+version: 2.1
 jobs:
+  "build-arm64":
+    machine:
+      enabled: true
+      image: ubuntu-2004:202101-01
+    resource-class: arm.medium
```
Comment (FedeDP) [inline .circleci/config.yml:7]: This should be `resource_class`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -357,11 +381,13 @@ jobs:
           command: |
             FALCO_VERSION=$(cat /build/release/userspace/falco/config_falco.h | grep 'FALCO_VERSION ' | cut -d' ' -f3 | sed -e 's/^"//' -e 's/"$//')
             /source/falco/scripts/publish-rpm -f /build/release/falco-${FALCO_VERSION}-x86_64.rpm -r rpm-dev
+            /source/falco/scripts/publish-rpm -f /build/release/falco-${FALCO_VERSION}-aarch64.rpm -r rpm-dev
```
Comment (FedeDP) [inline .circleci/config.yml:384]: The same lines must be added also to `publish/packages` and `publish/packages-deb` job, again except for the bin static one.

**Comment by FedeDP** [comment]
So, we surely need to port  `publish-deb` script.
As you can see, `publish-bin` already takes architecture as argument. I think we must do the same with `publish-deb`:
basically, on this line: https://github.com/falcosecurity/falco/blob/master/scripts/publish-deb#L41 you must check the external argument provided instead of declaring a local variable.

`publish-rpm` does not seem to reference `x86_64` or any `uname` variable, therefore i guess we shouldn't touch it.

I will need someone else to review the workflow too :) It seems fine to me:
## Build and test workflow
* we have the `build-arm64` job that builds and stores artifacts for arm64
* `build-arm64` is required by `rpm-sign` that is itself required by `publish/packages-dev`
* `build-arm64` is also required by `publish/packages-deb-dev`

## Release workflow
* we have the `build-arm64` job that builds and stores artifacts for arm64
* `build-arm64` is required by `rpm-sign` that is itself required by `publish/packages`
* `build-arm64` is also required by `publish/packages-deb`

**Comment by FedeDP** [comment]
To maintainers: note that the `build-arm64` job will never work unless the `falcosecturity/falco-builder:latest` image is available for arm64; that's part of #1990!

**Comment by FedeDP** [comment]
@odidev it seems like the new v2.1 workflows version is breaking the build:
> # [#/jobs/driver-loader~1integration] string [driver-loader/integration] does not match pattern ^[A-Za-z][A-Za-z\\s\\d_-]*$

Can you revert to 2?

**Comment by FedeDP** [comment]
Given that this PR depends upon #1990 , if you wish i can take over the `publish` scripts part; then you will need to rebase this one, once my PR is reviewed and eventually merged.
WDYT?

**Comment by odidev** [comment]
> Can you revert to 2? 

Usage of arm resources makes it mandatory to use workflow 2.1 . I checked out the error its nothing major, changing the name of the job should do the trick, I think it would better to mark this as WIP as it still might need some more modifications based on version 2.1 which can be figured only when we are able to run it. We can focus on releasing docker images and binary for arm64 for now. 

> Moreover, can you set "Fixes [#1891](https://github.com/falcosecurity/falco/issues/1891)" in the OP? 

The issue is for getting testing enabled for arm64, currently we are skipping it. Once we are able to add it too we can probably close it.

> Given that this PR depends upon [#1990](https://github.com/falcosecurity/falco/pull/1990) , if you wish i can take over the publish scripts part; then you will need to rebase this one, once my PR is reviewed and eventually merged. 
WDYT? 

Sure thing.

**Comment by FedeDP** [comment]
> The issue is for getting testing enabled for arm64

Mmh aren't we actually testing it? https://github.com/falcosecurity/falco/pull/1997/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R14

> Usage of arm resources makes it mandatory to use workflow 2.1

Oh i see thanks! Yeah renaming the job should do the trick; i think you can proceed with that!

**Comment by FedeDP** [comment]
@odidev i updated #1990 fixing publishing (deb,rpm) scripts.
They must be used like this:
```
/source/falco/scripts/publish-rpm -f /build/release/falco-${FALCO_VERSION}-x86_64.rpm -f /build/release/falco-${FALCO_VERSION}-aarch64.rpm -r rpm

/source/falco/scripts/publish-deb -f /build/release/falco-${FALCO_VERSION}-x86_64.deb -f /build/release/falco-${FALCO_VERSION}-aarch64.deb -r deb-dev
```

(notice that passing both packages in a single call is needed!)
Can you update your PR accordingly?
Thanks!

**Comment by odidev** [comment]
> Mmh aren't we actually testing it? https://github.com/falcosecurity/falco/pull/1997/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47R14 

Ah ok updated. 

> Oh i see thanks! Yeah renaming the job should do the trick; i think you can proceed with that! 

I have changed the name of the job and validated configuration file using CircleCI command line tool as well. 
**Note:** I had to change name of all jobs as `/` was present in the name of jobs which generates error in workflow 2.1 as CircleCI that it is not able to find orb of the name same as letters before the `/`. 

> (notice that passing both packages in a single call is needed!) 
Can you update your PR accordingly? 
Thanks! 

I have updated it accordingly.

**Comment by FedeDP** [comment]
Wow thank you! This PR should be ready once my #1990 is merged!
Thank you for your really hard work!!

**Comment by odidev** [comment]
Since this PR depends on #1990 , by when can we expect it to get merged?

**Comment by jasondellaluce** [comment]
#1990 has been merged! 🥳 @odidev, would you proceed and rebase this on master when you have some spare time? Looking forward to see this getting merged too!

**Comment by leogr** [comment]
Hey @odidev 

I'm going to rebase your PR, because we need it to fix the CI :) 
I'm also going to rebuild and push the `falcosecurity/falco-builder` image which is need to be updated to build for arm64, as reported by @FedeDP in [this comment](https://github.com/falcosecurity/falco/pull/1997#issuecomment-1122544714).

I'll keep you posted.

**Comment by leogr** [comment]
Quick update: we're trying to test the new `falcosecurity/falco-builder` with this PR. So, sorry if we're spamming you :sweat_smile: We hope are close to a fix.

**Comment by leogr** [comment]
Closing and reopening to trigger the CI
/close

**Comment by leogr** [comment]
Closing and reopening to update CI required checks
/close

**Pull request merged**